C++/source/string/permutations.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+	Permutations of a string
+	_________________________
+	
+	Print all the permutations of a string (including duplicates if
+	the input string contains repeating characters)
+
+	Time complexity
+	_______________
+	
+	O(N*N!)
+	Use of backtracking
+
+	Space complexity
+	________________
+	
+	O(N*N!)
+*/
+
+#include<iostream>
+#include<string.h>
+
+using namespace std;
+
+void permutations(string& a, int first, int last)
+{
+   	if(first==last)
+		cout<<a<<endl;
```

Comment:
Instead of printing all the permutations, how about passing a `std::set` by reference, and inserting permutations in the set. This will also eliminate duplicates as a `set` only stores a value once.

Also, this will make it easier to check the algorithm by writing tests that will check whether after the function is executed, the set contains the expected permutated strings for a given input string.